falcon/request.py

```diff
@@ -1075,6 +1075,7 @@ def _parse_form_urlencoded(self):
         # content length will only ever be read once per
         # request in most cases.
         body = self.stream.read()
+        self.stream.seek(0)
```

Comment:
Unfortunately, this won't always work; the WSGI server may pass a non-seekable file-like object to Falcon. One option would be to replace the WSGI object with our own that buffers the request body (similar to [my suggestion here][1]). This could be used as a stepping stone to the broader redesign around form param handling that is addressed by #493.

[1]: https://github.com/falconry/falcon/pull/493#discussion_r40822962